[MAINTENANCE] Add NYC taxi data to testing datasets

cdkini opened the pull request:
Changes proposed in this pull request:
- Addition of 2018-2020 NYC taxi data (as well as related script and Juptyer notebook)

---

Remarks:

- The data should include everything from 2018 - 2020 (with the first 3 months of 2019 coming from prior commits). Each CSV contains 10,000 rows and consistent column titles.
- Note that there are some rows that contain negative or NaN values; these are noted in the Jupyter Notebook but may play a role when using this data for unit/integration tests.

---

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-


### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

Commit:
test(prof): Add taxi data, script, & analysis

This commit contains the following:
- taxi_yellow_trip_data_samples: Uniform samples from the original datasets
- taxi_sampling_script: What was used to generate the above samples
- taxi_sampling_stats: Analyzes basic metrics found in the data

The data should include everything from 2018 - 2020 (with the first 3
months of 2018 coming from prior commits). Each CSV contains 10,000 rows
and equivalent column titles.

Note that there are some rows that contain negative or NaN values; these
are noted in the Jupyter Notebook but may play a role when using this
data for unit/integration tests.

cdkini commented:
@alexsherstinsky here's the PR I discussed with you earlier today. I could perhaps add a bit more to the Jupyter Notebook but figured it was prudent to open a PR and start a discussion first.

Commit:
style(misc): Run pre-commit hooks on datasets dir

cdkini commented:
Fixed linting issues after setting up proper pre-commit hooks.

Next step to get CI passing is to look into these docs integration tests that keep failing.

talagluck commented on tests/test_sets/taxi_yellow_trip_data_samples/taxi_sampling_script.py:
```diff
@@ -0,0 +1,90 @@
+            "passenger_count": pd.Int64Dtype(),
+            "RatecodeID": pd.Int64Dtype(),
+            "payment_type": pd.Int64Dtype(),
+        },
```
Nice!

talagluck reviewed (commented):
This is awesome! My one question is whether/where we want to include the scripts themselves in the main Great Expectations repo. In the past, I believe we've had repos containing internal tools, and I think this might be a good fit for something like that, unless we think this is something that will be used by general users of GE. Nice work!

cdkini commented:
I think it might make sense to move the script somewhere else since the sole purpose of this PR is to add data to aid our internal team's testing. As long as this data is accurate and usable, there should be little reason to have to replicate these efforts.

The Jupyter Notebook highlights some stats about the datasets, which could be useful when designing tests that use this information. Perhaps we remove the script but add reference to it in the notebook? This can ensure that future users of the data have the option of reviewing the details if deemed necessary.

Think it all depends on how much this data will be used :)

Commit:
test(fixtures): Update yellow trip fixtures

As we had added the entirety of 2018 - 2020 yellow trip datasets, we had
to account for the fact that "month" was no longer a unique identifier
for our data connector query.

To keep the size and scope of the tests the same, we specify the year to
be 2019.

Commit:
fix(tests): Fix Pandas and Snowflake test errs

Modified tests in our docusaurus integration dir to get them to pass
Azure CI. Previous iteration did not account for addition of 2018-2020
taxi datasets.

Commit:
chore(misc): Delete scripts used for taxi sampling

These scripts are to be moved to an internal repository since they do
not need to be shared with GE users.

Commit:
chore(misc): Move FIXME out of test matrix

Commit:
chore(misc): Update spacing of comment

talagluck commented on tests/integration/fixtures/yellow_trip_data_pandas_fixture/one_multi_batch_request_one_validator.py:
```diff
@@ -24,33 +24,33 @@
             batch_identifiers["month"]
         )
-        < 3
+        < 12
```
Not strictly necessary, but it might make things a bit clearer to specify the year here as well, something like
```suggestion
        < 12 and batch_identifiers["year"]=="2019"
```

cdkini replied:
I agree with this; explicitly stating the year just makes the condition that much clearer.

talagluck commented on tests/integration/test_script_runner.py:
```diff
@@ -69,22 +69,28 @@ class BackendDependencies(enum.Enum):
 ]
 
+"""
+FIXME(cdkini): Kept running into a sqlalchemy.exc.OperationalError when running Snowflake tests.
```
Tiny nit: people tend to use TODO instead of FIXME in the codebase

cdkini replied:
I'll change for consistency purposes!

talagluck reviewed (approved):
LGTM! Just a couple of possible suggestions, to your preference.

Commit:
chore(misc): Misc changes for clarification

Commit:
chore(misc): Fix lint issues

Commit:
Merge branch 'develop' into MAINTENANCE/GE-160/GE-298/add_nyc_taxi_datasets

Commit:
chore(misc): Fix lint
